tests/GridFS/StreamWrapperFunctionalTest.php

```diff
@@ -169,6 +169,29 @@ public function testWritableStreamStat()
         $stat = fstat($stream);
         $this->assertSame(6, $stat[7]);
         $this->assertSame(6, $stat['size']);
+  }
+
+    public function testWritableStreamStatAfterSaving()
+    {
+        $id = new ObjectId;
+        $stream = $this->bucket->openUploadStream('filename', ['chunkSizeBytes' => 1024, '_id' => $id]);
+
+        $this->assertSame(6, fwrite($stream, 'foobar'));
+        $this->assertTrue(fclose($stream));
+        $currentTimestamp = time();
+        $stream = $this->bucket->openDownloadStream($id);
+
+        $stat = fstat($stream);
+        $this->assertSame(0100444, $stat[2]);
+        $this->assertSame(0100444, $stat['mode']);
+        $this->assertSame(6, $stat[7]);
+        $this->assertSame(6, $stat['size']);
+        $this->assertGreaterThanOrEqual($currentTimestamp, $stat[9]);
+        $this->assertGreaterThanOrEqual($currentTimestamp, $stat['mtime']);
+        $this->assertGreaterThanOrEqual($currentTimestamp, $stat[10]);
+        $this->assertGreaterThanOrEqual($currentTimestamp, $stat['ctime']);
```

Comment:
This can use `assertSame()` if using the `length-10` file, since we'll know the exact value of `uploadDate`.

Note: 1484202200000 is milliseconds, so that will need to be converted to seconds for these values.